### PR TITLE
docs: use v[x.y] instead of v[x.y.z]

### PR DESCRIPTION
**Pull request by zirain**
follow up https://github.com/envoyproxy/gateway/pull/3819

we should not create docs for each patch.

### CODE REVIEW

File: tools/make/docs.mk [hunk 1]
```diff
@@ -101,8 +101,9 @@ docs-release-prepare:
 .PHONY: docs-release-gen
 docs-release-gen:
 	@$(LOG_TARGET)
-	@$(call log, "Added Release Doc: site/content/en/$(TAG)")
-	cp -r site/content/en/latest site/content/en/$(TAG)
+	$(eval DOC_VERSION := $(shell cat VERSION | cut -d "." -f 1,2))
+	@$(call log, "Added Release Doc: site/content/en/$(DOC_VERSION)")
+	cp -r site/content/en/latest/ site/content/en/$(DOC_VERSION)/
 	@for DOC in $(shell ls site/content/en/latest/user); do \
```
Comment (zirain) [inline tools/make/docs.mk:107]: this had been broken for a long time, https://github.com/envoyproxy/gateway/pull/3766 will help.

**Review by arkodg — APPROVED** [review]
LGTM thanks !

**Review by zhaohuabing — APPROVED** [review]
🚀

**Comment by zirain** [comment]
Let's merge this and see what will happen.

**Pull request merged**